### DISCUSSION TITLE
Have the custom tag follow whatever it's renamed to.

### DISCUSSION THREAD

**Discussion by xahodo** [vkbo/novelWriter#1427] [Ideas]
There's the `custom` tag. However seeing that you can rename its label, wouldn't it be better to have the custom tag follow the label of whatever it's renamed to? People could have multiple custom tags, each with their own name. This could cause confusion for them if I had the custom tags `foo` and `bar`, they would both be considered `custom` tags by NovelWriter, being cause of confusion.

What I would like to see:

Create new custom tag, and name it "foo", from that moment onward I would address it with `@foo:` in my documents/notes.
Then I'd create another custum tag, and name it "bar", from that moment I would address it as `@bar:` in my documents/notes.
They would show up in the project tree as "foo" and "bar", and so they need to be able to be referenced as such.

I hope I am clear enough. Sorry if I'm unclear about what I mean.

**Comment by vkbo** [comment]
This is related to the discussion in #1399.

If you use the custom category extensively, you can solve most of this by just prefixing your tags. I'm planning a revamp of the tag/reference system down the line, but I haven't decided what would be the best approach yet. I'll certainly keep this in mind. You could perhaps also post this as a feature request. Your text is pretty clear, so feel free to just copy/paste it.

**Reply by wessleyn** [reply]
Why ain't we just allowing the use to define the tag name for each document . Cause I am having a hard time referencing my notes in my scenes

I.e I group a novel into volumes - milestones - scenes  - chapters

And I want to be able to ref  like this

This might be for unorganized scenes or chapters in the novel root dir


@volume: name of volume


For individual volumes 
@milestone


There's this feature where a novel note / doc can be used as a folder meaning where I would have the folder name , they'be maybe a synopsis and tags referencing the milestone 

And so on...

**Reply by vkbo** [reply]
Firstly, please don't tag random users on GitHub in your posts. They get spammed with notifications if you do.

Further, I don't really follow what you're suggesting here. You can already reference parts of your plot with the `@plot` reference keyword. And as mentioned, you can structure the tags themselves into whatever way you want with folders and prefixes of the tags.

The reason why the reference keywords are predefined is because they are linked to the respective root folders.

I've noticed from a number of feature requests that some people use the tags and reference system in ways it wasn't really designed for, and have introduced a considerable complexity of usage. What it is intended for, originally, is to tag each scene with which characters are in it, the point of view, and the scene location. It is based on the principles used in screenwriting.
